### ISSUE TITLE
expected dense_1 to have shape (6,) but got array with shape (9,) Error

### ISSUE THREAD

**Issue by ghazalbargshady** [keras-team/keras#11439]
Hi
I got an error in running my code. anyone can help?

Error: expected dense_1 to have shape (6,) but got array with shape (9,)
The role of code to classification images in 6 classes.
code:
PSPI_folder='.../PSPI/042-ll042'
img_folder='.../Images/042-ll042'
X=[]
Y=[]
folders=os.listdir(PSPI_folder)
for i in range(len(folders)):
    foldername=folders[i]
    PSPI_folder1=os.path.join(PSPI_folder,foldername)
    img_folder1=os.path.join(img_folder,foldername)
    imgfiles=os.listdir(img_folder1)
    pspifiles=os.listdir(PSPI_folder1)
    for j in range(len(pspifiles)):
        file_name=pspifiles[j]
        PSPIfile_path=os.path.join(PSPI_folder1,pspifiles[j])
        imgfile_path=os.path.join(img_folder1,imgfiles[j])
        PSPIfile = open(PSPIfile_path, "r")
        cs=int(float(str(PSPIfile.read()).replace('\n','')))
        Y.append(cs)
        img=cv2.imread(imgfile_path)
        img.resize(224,224,3)
         v = np.array(img)
        im = img.reshape(-1,1,224,224,3)
        v = np.array(im)
        v0=[]
        
        for vv in v:
            v0.append(vv[0])
        X.append(np.array(v0))
        
def get_y(Y):
    output=[]
    n_tags = np.max(Y) + 1
    for i in range(len(Y)):
        vv=[]
        for k in range(n_tags):
            if(k==Y[i]):
                vv.append(1)
            else:
                vv.append(0)
        output.append(vv)
    return output
X=np.array(X)
Y=np.array(Y)
n_tags=np.max(Y)+1
YY=np.array(get_y(Y))
X_tr, X_te, y_tr, y_te = train_test_split(X, YY, test_size=0.01)
print (img.shape)

print (X_tr.shape)
print (X_tr.shape)
vggface_model = VGGFace(include_top=False,model='resnet50',input_shape=(224, 224, 3))
top_layers_checkpoint_path = 'cp.top.best.hdf5'
fine_tuned_checkpoint_path = 'cp.fine_tuned.best.hdf5'
last_layer = vggface_model.get_layer('avg_pool').output
x = Flatten(name='flatten')(last_layer)
vfmodel_out = Dense(6, activation='softmax', name='classifier')(x)
vfmodel = Model(input=vggface_model.input, output=vfmodel_out)

# first: train only the top layers (which were randomly initialized)
# i.e. freeze all vggface layers
for layer in vfmodel.layers:
    layer.trainable = False

#our model

input_lay = Input(shape=(None, 224, 224, 3)) #dimensions of your data
time_distribute = TimeDistributed(Lambda(lambda x: vfmodel(x)))(input_lay)
lstm_lay = LSTM(20)(time_distribute)
output_lay = Dense(6, activation='relu')(lstm_lay)
model = Model(inputs=[input_lay], outputs=[output_lay])

# compile the model (should be done *after* setting layers to non-trainable)
model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['categorical_accuracy'])

#train the model on the new data for a few epochs
model.fit(X_tr, y_tr,  batch_size=24,
          epochs=1,
          validation_data=(X_te, y_te))

**Comment by bp274** [comment]
I was having a similar error which i later found out was because of the [input parameter to model.fit](https://keras.io/models/model/) not being a numpy array. Here i think the problem is that the [train_test_split returns a list](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html), which is not a suitable parameter type for fit.

**Comment by pachpandepriti** [comment]
@ghazalbargshady Did you solve the problem?

**Comment by ghazalbargshady** [comment]
thanks
